### PR TITLE
next/635/20241112/v1

### PR DESCRIPTION
**Pull request by victorjulien**
Staging:
 - #12045
 - #12109
 - #12114

### CODE REVIEW

**Review by jasonish — APPROVED** [review]

**Comment by victorjulien** [comment]
Thanks. Waiting for CI and QAlab results.

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .app_layer.flow.dcerpc_tcp | 40 | 42 | 105.0% | 

Pipeline 23310

**Comment by jufajardini** [comment]
> Thanks. Waiting for CI and QAlab results.

May I also request a re-review of the signature types table, before merge? Columns `Inspected` and `Matches`, especially.

File: doc/userguide/rules/intro.rst
Comment (victorjulien) [inline doc/userguide/rules/intro.rst:559]: this described pre-7 behavior I think, in 7+ this is pkt, see https://github.com/OISF/suricata-verify/pull/2066/files#diff-d0226a73eb0338d3e349b8142fcc97004510153ca21168e6d911bd8b2e48f6faR14

**Comment by victorjulien** [comment]
2025200 is the only open rule I see with `app_layer`, but I don't yet see how it is different other than the threshold keyword.

File: doc/userguide/rules/intro.rst
Comment (jufajardini) [reply doc/userguide/rules/intro.rst:559]: Then I run a test just like that, it still renders as 'app_layer to me'. Only if we add `flow:established` do I see the change in behavior.

Comment (jufajardini) [reply doc/userguide/rules/intro.rst:559]: But I think this should be mentioned in the docs.

Comment (victorjulien) [reply doc/userguide/rules/intro.rst:559]: Yes, I just noticed the same:
```json
["pkt","pass tls any any -> any 443 (flow:established; sid:5;)"]
["pkt","pass tcp any any -> any 443 (flow:established; app-layer-protocol:tls; sid:6;)"]
["app_layer","alert http any any -> any any (sid:1;)"]
["app_layer","alert http any any -> any any (sid:2; threshold:type limit, track by_src, count 1, seconds 60;)"]
["pd_only","alert tcp any any -> any any (sid:3; app-layer-protocol:http;)"]
["pd_only","alert tcp any any -> any any (sid:4; app-layer-protocol:http; threshold:type limit, track by_src, count 1, seconds 60;)"]
```

Comment (jufajardini) [reply doc/userguide/rules/intro.rst:559]: I think we should add such examples to the `rule-types` set of SV tests, and have a v8 for the docs with a note about this.

**Comment by victorjulien** [comment]
Doc PR needs another revision.

**Pull request closed without merging**
